doc/Data Plane Code Generation/Subclients.md

```diff
@@ -0,0 +1,233 @@
+# Customizing the generated code
+
+## Subclients
+
+There are two categories of clients: service clients and their subclients. Service clients can be instantiated and have the Client suffix. Subclients can only be created by calling factory methods on other clients (commonly on service clients) and do not have the client suffix. The service client is the entry point to the API for an Azure service. Please refer to [guideline](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-subclients) for more detail information.
+
+There are two tools to organize subclients and service client.
+
+- **Single top-level client**: define one top-level service client with the name <service_name>Client, all other clients will be its subclient.
+
+- **Client hierarchy**: organize the client service and subclients via defining the parent-subclient hierarchy dependency.
+
+**Note**: You can combine those two approaches to define an client hierarchy with one top-level client and its subclient may have sub clients also. 
+
+### Single top-level client
+
+With multiple service clients it may be better to group them all under one top-level client.
+
+<details>
+
+**Generated code before:**
+
+``` C#
+//Generated\Client1Client.cs
+namespace Azure.Service.SubClients
```

Comment:
Can we use a real service like Purview for this case?  I feel a bit difficult to understand the concept here like
the name space is call Azure.Service.SubClients, but the Client1Client should be a service client. And that name is also a bit strange

Reply:
same as above